### PR TITLE
Add a repl as an included binary

### PR DESCRIPTION
**Pull request by aturon**
Pretty straightforward, but really helpful for exploration.

### CODE REVIEW

**Comment by aturon** [comment]
cc @withoutboats, I bet you'll find this useful too!

File: src/bin/repl.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+use std::sync::Arc;
+
+use chalk::ir;
+use chalk::lower::*;
+use chalk::solve::prove::Prove;
+use chalk::solve::solver::Solver;
+
+use rustyline::error::ReadlineError;
+
+error_chain! {
```
Comment (nikomatsakis) [inline src/bin/repl.rs]: I ❤️ `error_chain`

cc @brson :)

Comment (aturon) [inline src/bin/repl.rs]: Same here! We should consider it for the nursery, tbh...

**Comment by withoutboats** [comment]
Haha awesome! I had definitely thought about this; in the long term it seems like one of the best uses for chalk.

**Comment by nikomatsakis** [comment]
rustup seems to be broken with travis :(

**Comment by aturon** [comment]
@nikomatsakis there's a rustup outage for nightly right now. should resolve in a day's time.

**Comment by nikomatsakis** [comment]
Forgot about this.

**Pull request merged**